ThreeDSecure/src/main/java/com/braintreepayments/api/CardinalClient.java

```diff
@@ -65,6 +69,36 @@ private void configureCardinal(Context context, Configuration configuration, Thr
         cardinalConfigurationParameters.setEnvironment(cardinalEnvironment);
         cardinalConfigurationParameters.setRequestTimeout(8000);
         cardinalConfigurationParameters.setEnableDFSync(true);
+
+        switch (request.getUiType()) {
+            case 1:
```

Comment:
There should be a way to reference the `@IntDef` types in the switch statement. However [BraintreeRequestCodes.java](https://github.com/braintree/braintree_android/blob/master/BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeRequestCodes.java) is set up, we're able to reference the enum constants.

Reply:
Since these are new we could go with enums. `@IntDef` was made back in the day as an optimization because enums were really expensive for some reason on Android, but they [recently fixed this](https://medium.com/trade-me/android-then-and-now-intro-intdef-enums-bca22d5cca56). Next major version we'll convert everything to enum.

Reply:
Nice! I didn't know this was possible. That's dope. Updated here: https://github.com/braintree/braintree_android/pull/738/commits/fd649dc4d3e2f82ccb9c7badef2f1cc3d24718d7